video/graphics.h

```diff
@@ -100,6 +100,44 @@ RGB888 getPixel(uint16_t x, uint16_t y) {
 	return RGB888(0,0,0);
 }
 
+// Horizontal scan for non-equals pixel
+uint16_t scanH(int16_t x, int16_t y, RGB888 colour, int8_t direction = 1) {
```

Comment:
these routines are inefficient and slow.  the main issue with them is the calls to `canvas->getPixel` which will be grabbing an individual pixel and converting it to RGB888 for comparison against an RGB888 colour.

a better way to do this would be inside the appropriate VGAControllers inside vdp-gl.  this would allow for more direct pixel checks to be done without all this conversion, and would result in an order of magnitude improvement in speed

(excessive debugging in these routines will be got rid of)

Reply:
debugging within these two scan functions has been removed. the  `fillHorizontalLine` function still logs